javascript/code-challenges/challenges-01.test.js

```diff
@@ -0,0 +1,222 @@
+    newArr.push(newArray);
+  });
+  return newArr;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named `greeting` that takes in a single string and returns the string in all uppercase letters, and followed by an "!".
+
+Then, write a function named `speaker` that takes in an array of strings and a callback function. 
+
+Use `forEach` to build a new array of strings, each string modified by the callback. Return the new array. 
+------------------------------------------------------------------------------------------------ */
+
+const greeting = (word) => {
+  // Solution code here...
+
+  return word.toUpperCase() + '!';
+};
+
+
+const speaker = (words, callback) => {
+  // Solution code here...
+  let newArr = [];
+
+  words.forEach(arr => {
+    callback(arr);
+    newArr.push(callback(arr));
+  });
+  return newArr;
+};
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named addValues that takes in an array and a value and pushes the value into the array. This function does not need a return statement.
+
+Then, write a function named addNumbers that takes in four arguments:
+  - A number to be added to an array
+  - An array into which the number should be added
+  - The number of times the number should be added
+  - A callback function to use to add the numbers to the array (Hint: you already defined it)
+
+Within the addNumbers function, invoke the callback function as many times as necessary, based on the third argument of the addNumbers function.
+
+Return the modified array.
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr, value) => {
+  // Solution code here...
+  let newArr = [];
+  newArr.push(value);
+};
+
+const addNumbers = (num, arr, times, callback) => {
+  // Solution code here...
+  arr.forEach.push(num);
+
+};
+
+/* ------------------------------------------------------------------------------------------------
+
+CHALLENGE 6
+
+Write a function named createList that takes in an array of the current store inventory.
+
+The inventory is formatted like this:
+[
+  { name: 'apples', available: true },
+  { name: 'pears', available: true },
+  { name: 'oranges', available: false },
+  { name: 'bananas', available: true },
+  { name: 'blueberries', available: false }
+]
+
+This function should use forEach to populate your grocery list based on the store's inventory. If the item is available, add it to your list. Return the final list.
+------------------------------------------------------------------------------------------------ */
+
+const createList = (availableItems) => {
+  // Solution code here...
+  let newArr = [];
+
+  availableItems.forEach( () => {
+    if (inventory[1] === true){
+      newArr.push(name);
+    }
+  });
+  return newArr;
+};
```

Comment:
You're almost here with this one, we just need to consider how we can know if an item is available or not before we push the name to the `newArr`.

In this case, we have an object and we are using forEach to look at each `item`. )n line 138 we need to pass in a callback --> `availableItems.forEach(item =>)`

From there, if the item is available, only then will we push the name of the item. (For objects, we have to consider how to access the keys of  `name` and `available`. See here[Property Accessors](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_accessors)